<launch>

	<param name="use_sim_time" value="false" />

	 <node name="mapping" pkg="cob_3d_experience_mapping" type="cob_3d_experience_mapping_lemon_cloud_node" output="screen" >
		<param name="step_mode" value="false" />
		<param name="cloud_server" value="localhost:12347" />
		
		<remap from="/odom" to="/robot0/odom" />
		<remap from="/PoseCell/TopologicalAction" to="/irat_red/PoseCell/TopologicalAction" />
	 </node>
	 
	  <node name="sim_features" pkg="cob_3d_experience_mapping" type="sim_features.py" output="screen">
		<param name="NUM_FEATURES" value="-1" />
	  </node>

	<node name="RatSLAMExperienceMap" pkg="ratslam_ros" type="ratslam_em" args="$(find ratslam_ros)/config/config_irataus.txt _image_transport:=compressed" cwd="node" required="true">
		<remap from="/irat_red/odom" to="/robot0/odom" />
	</node>
	 
	<node pkg="rosbag" type="play" name="play" args="$(find cob_3d_experience_mapping)/evaluation/datasets/office_odom.bag --clock -d 0.5 -r 2 -s 0">
	</node>
	
	

	 <node name="mapping2" pkg="cob_3d_experience_mapping" type="cob_3d_experience_mapping_lemon_cloud_node" output="screen" >
		<param name="step_mode" value="false" />
		<param name="cloud_server" value="localhost:12347" />
		
		<remap from="/odom" to="/robot0/odom2" />
		<remap from="/sim_barks/sensor_info" to="/sim_barks/sensor_info2" />
		<remap from="/PoseCell/TopologicalAction" to="/irat_red/PoseCell/TopologicalAction2" />
	 </node>
	 
	  <node name="sim_features2" pkg="cob_3d_experience_mapping" type="sim_features.py" output="screen">
		<param name="NUM_FEATURES" value="-1" />
		<remap from="/robot0/odom" to="/robot0/odom2" />
		<remap from="/sim_barks/sensor_info" to="/sim_barks/sensor_info2" />
	  </node>
	  
	<node name="RatSLAMExperienceMap2" pkg="ratslam_ros" type="ratslam_em" args="$(find ratslam_ros)/config/config_irataus.txt _image_transport:=compressed" cwd="node" required="true">
		<remap from="/irat_red/odom" to="/robot0/odom2" />
		<remap from="/irat_red/PoseCell/TopologicalAction" to="/irat_red/PoseCell/TopologicalAction2" />
	</node>
	 
	<node pkg="rosbag" type="play" name="play2" args="$(find cob_3d_experience_mapping)/evaluation/datasets/office_odom.bag -d 10.5 -r 2 -s 0">
		<remap from="/robot0/odom" to="/robot0/odom2" />
	</node>
</launch>
